packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -111,79 +89,87 @@ export class ProjectService {
-        ]);
-        const adapter = await this.jobModel.tryJobStep(
-            jobUuid,
-            JobStepType.TESTING_ADAPTOR,
-            async () => ProjectService.testProjectAdapter(data),
-        );
-        const explores = await this.jobModel.tryJobStep(
-            jobUuid,
-            JobStepType.COMPILING,
-            async () => adapter.compileAllExplores(),
-        );
+            const projectUuid = await this.jobModel.tryJobStep(
+                job.jobUuid,
+                JobStepType.CREATING_PROJECT,
+                async () =>
+                    this.projectModel.create(user.organizationUuid, data),
+            );
+            await this.projectModel.saveExploresToCache(projectUuid, explores);
 
-        const projectUuid = await this.jobModel.tryJobStep(
-            jobUuid,
-            JobStepType.CREATING_PROJECT,
-            async () => this.projectModel.create(user.organizationUuid, data),
-        );
+            // do we still need project adapters?
+            this.projectAdapters[projectUuid] = adapter;
 
-        await this.jobModel.tryJobStep(
-            jobUuid,
-            JobStepType.CACHING,
-            async () => {
-                await this.projectModel.saveExploresToCache(
+            await this.jobModel.update(job.jobUuid, {
+                jobStatus: JobStatusType.DONE,
+                jobResults: {
                     projectUuid,
-                    explores,
-                );
-            },
-        );
+                },
+            });
 
-        await this.jobModel.upsertJobStatus(
-            jobUuid,
-            projectUuid,
-            JobStatusType.DONE,
-        );
+            analytics.track({
+                event: 'project.created',
+                userId: user.userUuid,
+                properties: {
+                    projectName: data.name,
+                    projectId: projectUuid,
+                    projectType: data.dbtConnection.type,
+                    warehouseConnectionType: data.warehouseConnection.type,
+                    organizationId: user.organizationUuid,
+                    dbtConnectionType: data.dbtConnection.type,
+                },
+            });
+            this.projectAdapters[projectUuid] = adapter;
+        };
 
-        analytics.track({
-            event: 'project.created',
-            userId: user.userUuid,
-            properties: {
-                projectName: data.name,
-                projectId: projectUuid,
-                projectType: data.dbtConnection.type,
-                warehouseConnectionType: data.warehouseConnection.type,
-                organizationId: user.organizationUuid,
-                dbtConnectionType: data.dbtConnection.type,
-            },
-        });
-        this.projectLoading[projectUuid] = false;
-        this.projectAdapters[projectUuid] = adapter;
-        return this.getProject(projectUuid, user);
+        await this.jobModel.create(job);
```

Comment:
should we here check the lock ? or maybe it is not needed because it is not supposed to happen on create (could happen though)

Reply:
ok, forget it, because the lock is per project this should never happen